CryptKeeperZK/crypt-keeper-extension
# feat: backup rollback

## Explanation

This PR supports rollback storage for unsuccessful backup uploads.

Details are below:
- [x] Add rollback for unsuccessful backup uploads
- [x] Add restore/download storage methods for backupables
- [x] Add e2e tests for invalid backup uploads

## More Information

Closes #330 

## Screenshots/Screencaps

N/A

## Manual Testing Steps

1. Try to upload any invalid backup file from `packages/e2e/backups`.
2. Check errors
3. Try to upload valid backup file or create account with password
4. Check lock/unlock functions

## Pre-Merge Checklist


> PR template source from [github.com/MetaMask](https://github.com/MetaMask)
